File: womenwhogo/sandwich/sandwich.go
package main

import (
	"fmt"
	"time"
)

func main() {
	startTime := time.Now()
	pb := getPeanutButter()
	j := getJelly()
	makeSandwich(pb, j)
	fmt.Printf("Sandwich took %v to make\n", time.Since(startTime))
}

func makeSandwich(ingredientOne, ingredientTwo string) {
	fmt.Println(
		"putting together " + ingredientOne + " and " + ingredientTwo,
	)
}

func getPeanutButter() string {
	fmt.Println("Butler 1 says: I am getting the peanut butter")
	simulateWork()
	return "peanut butter"
}

func getJelly() string {
	fmt.Println("Butler 1 says: I am getting the jelly")
	simulateWork()
	return "jelly"
}

func simulateWork() {
	time.Sleep(time.Second)
}

File: womenwhogo/sandwich/sandwich.rb
class Butler 
    def get_peanut_butter
        sleep 1
        return "peanut butter"
    end

    def get_jelly
        sleep 1
        return "jelly"
    end
end

b1 = Butler.new
b2 = Butler.new

t1 = Time.new
pb = b1.get_peanut_butter
j = b2.get_jelly
t2 = Time.new
puts "took #{t2 - t1} seconds to get the ingredients"